File: node/src/main/java/dev.httpmarco.polocloud.node/terminal/commands/HelpCommand.java
package dev.httpmarco.polocloud.node.terminal.commands;

import dev.httpmarco.polocloud.node.Node;
import dev.httpmarco.polocloud.node.commands.Command;
import lombok.extern.log4j.Log4j2;

@Log4j2
public class HelpCommand extends Command {

    public HelpCommand() {
        super("help", "Help you with all commands", "?");

        defaultExecution(context -> {
            for (var command : Node.instance().commandService().commands()) {
                var aliases = command.aliases().length != 0 ? " &8(&7" + String.join("&8, &7", command.aliases()) + "&8)" : "";

                log.info("&f{}{} &8- &7{}&8.", command.name(), aliases, command.description());
            }
        });
    }
}

File: agent/src/main/java/dev/httpmarco/polocloud/agent/runtime/local/terminal/commands/impl/HelpCommand.kt
package dev.httpmarco.polocloud.agent.runtime.local.terminal.commands.impl

import dev.httpmarco.polocloud.agent.i18n
import dev.httpmarco.polocloud.agent.logger
import dev.httpmarco.polocloud.agent.runtime.local.terminal.commands.Command
import dev.httpmarco.polocloud.agent.runtime.local.terminal.commands.CommandService

class HelpCommand(private val commandService: CommandService) : Command("help", "Show all available commands", "?") {

    init {
        defaultExecution {
            i18n.info("agent.terminal.command.help.info")
            commandService.commands.forEach {
                logger.info(" &8- &f${it.name}&8: &7${it.description}")
            }
        }
    }
}